src/lib.rs

```diff
@@ -177,21 +193,35 @@ impl Instance {
     Ok(Instance { inst })
   }
 
+  // Work for any number of variables
+  fn pad_variables(&self, vars: &Vec<Scalar>) -> Vec<Scalar> {
+    let vars_pad = self.inst.get_num_vars() - vars.len();
+    let mut padding = vec![Scalar::zero(); vars_pad];
+    let mut padded_assignment = vars.clone();
+    padded_assignment.append(&mut padding);
+    padded_assignment
+  }
+
   /// Checks if a given R1CSInstance is satisfiable with a given variables and inputs assignments
   pub fn is_sat(
     &self,
     vars: &VarsAssignment,
     inputs: &InputsAssignment,
   ) -> Result<bool, R1CSError> {
-    if vars.assignment.len() != self.inst.get_num_vars() {
```

Comment:
Why are these checks removed. Couldn't someone call is_sat() with incorrect-sized VarsAssignment and InputsAssignment?

Reply:
The only offending case is vars.assignment.len() > inst_get_num_vars(), I added a check for that.